addons/binding/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/handler/RobonectHandler.java

```diff
@@ -0,0 +1,466 @@
+            return UnDefType.UNDEF;
+
+        }
+    }
+
+    private void refreshVersionInfo() throws InterruptedException {
+        VersionInfo info = robonectClient.getVersionInfo();
+        if (info.isSuccessful()) {
+            Map<String, String> properties = editProperties();
+            properties.put(Thing.PROPERTY_SERIAL_NUMBER, info.getRobonect().getSerial());
+            properties.put(Thing.PROPERTY_FIRMWARE_VERSION, info.getRobonect().getVersion());
+            properties.put(PROPERTY_COMPILED, info.getRobonect().getCompiled());
+            properties.put(PROPERTY_COMMENT, info.getRobonect().getComment());
+            updateProperties(properties);
+        } else {
+            logger.debug("Could not retrieve mower version info. Robonect error response message: {}",
+                    info.getErrorMessage());
+        }
+    }
+
+    private void refreshLastErrorInfo() throws InterruptedException {
+        ErrorList errorList = robonectClient.errorList();
+        if (errorList.isSuccessful()) {
+            if (errorList.getErrors() != null && errorList.getErrors().size() > 0) {
+                ErrorEntry lastErrorEntry = errorList.getErrors().get(0);
+                updateLastErrorChannels(lastErrorEntry);
+            }
+        } else {
+            logger.debug("Could not retrieve mower error list. Robonect error response message: {}",
+                    errorList.getErrorMessage());
+        }
+    }
+
+    private void updateLastErrorChannels(ErrorEntry error) {
+        if (error.getErrorMessage() != null) {
+            updateState(CHANNEL_LAST_ERROR_MESSAGE, new StringType(error.getErrorMessage()));
+        }
+        if (error.getErrorCode() != null) {
+            updateState(CHANNEL_LAST_ERROR_CODE, new DecimalType(error.getErrorCode().intValue()));
+        }
+        if (error.getDate() != null) {
+            State dateTime = convertDateTimeType(error.getDate(), error.getTime());
+            updateState(CHANNEL_LAST_ERROR_DATE, dateTime);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            RobonectConfig robonectConfig = getConfigAs(RobonectConfig.class);
+            RobonectEndpoint endpoint = new RobonectEndpoint(robonectConfig.getHost(), robonectConfig.getUser(),
+                    robonectConfig.getPassword());
+            httpClient = new HttpClient();
+            httpClient.start();
+            robonectClient = new RobonectClient(httpClient, endpoint);
+            refreshVersionInfo();
+            Runnable runnable = new MowerChannelPoller(robonectConfig);
+            int pollInterval = robonectConfig.getPollInterval();
+            pollingJob = scheduler.scheduleWithFixedDelay(runnable, 0, pollInterval, TimeUnit.SECONDS);
+        } catch (Exception e) {
+            logger.error("Exception when trying to initialize", e);
+            updateStatus(ThingStatus.OFFLINE);
```

Comment:
ok, now just wrap the start() and refreshVersionInfo() methods.

Reply:
after refactoring the InterruptedException, just start() is in the try-catch block :-)

Reply:
Hmmm, looks like this was going a little too far: I started the binding at the exact point in time when my mover was outside of the wifi coverage. The thing stayed in UNINTIALIZED because of this exception:

```
2018-05-02 13:21:34.740 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.robonect.handler.RobonectHandler@519b4a71': Could not send command http://shawn.gdhsn.kaufleute.name/json?cmd=version
org.openhab.binding.robonect.internal.RobonectCommunicationException: Could not send command http://shawn.gdhsn.kaufleute.name/json?cmd=version
	at org.openhab.binding.robonect.internal.RobonectClient.sendCommand(RobonectClient.java:308)
	at org.openhab.binding.robonect.internal.RobonectClient.getVersionInfo(RobonectClient.java:317)
	at org.openhab.binding.robonect.handler.RobonectHandler.refreshVersionInfo(RobonectHandler.java:339)
	at org.openhab.binding.robonect.handler.RobonectHandler.initialize(RobonectHandler.java:392)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)[:1.8.0_152]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)[:1.8.0_152]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[:1.8.0_152]
	at java.lang.reflect.Method.invoke(Method.java:498)[:1.8.0_152]
	at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:153)[49:org.eclipse.smarthome.core:0.10.0.201804301608]
	at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:53)[49:org.eclipse.smarthome.core:0.10.0.201804301608]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_152]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[:1.8.0_152]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[:1.8.0_152]
	at java.lang.Thread.run(Thread.java:748)[:1.8.0_152]
Caused by: java.util.concurrent.ExecutionException: java.net.SocketTimeoutException: Connect Timeout
	at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)
	at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:110)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:676)
	at org.openhab.binding.robonect.internal.RobonectClient.sendCommand(RobonectClient.java:291)
```

The exception dropped trough to the framework. As a SocketTimeoutException (or even: IOException) is something expected, it should be rather caught and the thing set to OFFLINE/COMMUNICATION_ERROR in such case.

Reply:
ups, lucky me, that you tested this in the right moment :-)
I now moved the loadVersionInfo again in the polling thread, and reload it each time the thing went offline. That way it would also pick up new properties if you do a firmware upgrade (or more probably downgrade at the moment ;-) ), without the need to restart the binding.